pkg/internal/controller/controller.go

```diff
@@ -83,6 +130,13 @@ type Controller[request comparable] struct {
 	// startWatches maintains a list of sources, handlers, and predicates to start when the controller is started.
 	startWatches []source.TypedSource[request]
 
+	// didStartEventSourcesOnce is used to ensure that the event sources are only started once.
+	didStartEventSourcesOnce sync.Once
+
+	// warmupResultChan receives the result (nil / non-nil error) of the warmup method. It is
+	// consumed by the WaitForWarmupComplete method that the warmup has finished.
```

Comment:
I think there's something wrong or missing in this sentence

Reply:
removed that field altogether :)